lib/td/client/api/job.rb

```diff
@@ -201,29 +197,29 @@ def job_result_each(job_id, &block)
   # @param [Proc] block
   # @return [nil]
   def job_result_each_with_compr_size(job_id, &block)
-    get("/v3/job/result/#{e job_id}", {'format'=>'msgpack'}) {|res|
-      if res.code != "200"
+    upkr = MessagePack::Unpacker.new
+    infl = nil
+
+    get("/v3/job/result/#{e job_id}", {'format'=>'msgpack'}) {|res, chunk, current_total_chunk_size|
+      if res.code != 200
         raise_error("Get job result failed", res)
       end
 
-      res.extend(DirectReadBodyMixin)
-      if res.header['Content-Encoding'] == 'gzip'
+      # default to decompressing the response since format is fixed to 'msgpack'
+      if !infl && (res.header['Content-Encoding'].include? 'gzip')
```

Comment:
This is a check for the second and subsequent loops don't want to create a Zlib:Inflate instance.